### PR TITLE
Session3 - Assignment

### PR DESCRIPTION
**Pull request by angielynr**
Implemented EF Migration, Product Service & configurations.

session3

### CODE REVIEW

File: Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+
+            return product.Id;
+        }
+
+        public async Task UpdateProduct(UpdateProductDto productForUpdate)
+        {
+            var product = await _appDbContext.Products.FirstOrDefaultAsync(p => p.Id == productForUpdate.Id);
+            if (product == null)
+            {
+                throw new ArgumentException("Product not found.");
```
Comment (laira-salvador) [inline Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs]: You can handle the checking of the product in the controller before calling the UpdateProduct() in service

File: Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+            product.Name = productForUpdate.Name;
+            product.ShopId = productForUpdate.ShopId;
+
+            await _appDbContext.SaveChangesAsync();
+        }
+
+        public async Task DeleteProduct(int id)
+        {
+            var product = await _appDbContext.Products.FirstOrDefaultAsync(p => p.Id == id);
+            if (product == null)
```
Comment (laira-salvador) [inline Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs]: Same with the UpdateProduct

**Review by laira-salvador — CHANGES_REQUESTED** [review]
validate if objects is existing in the controller

File: Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs (hunk 1 above)
Comment (angielynr) [reply Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs]: Updated code. Thanks!

File: Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs (hunk 2 above)
Comment (angielynr) [reply Session3/EFCore - Assignment 3 Base/EFCoreAssignment.Services/ProductService.cs]: Updated code. Thanks!

**Review by laira-salvador — APPROVED** [review]
LGTM!

**Pull request merged**